pkg/api/v1/routes/handlers.go

```diff
@@ -212,8 +212,21 @@ func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
 		}
 	}
 
+	var cleanup func() error
+	var shouldCleanup bool
+	defer func() {
+		if !shouldCleanup {
+			return
+		}
+		if err := cleanup(); err != nil {
+			r.logger.WithFields(logrus.Fields{
+				"error": err,
+			}).Error("failed to cleanup server")
+		}
+	}()
 	// Creates a server record in FleetDB.
-	if err = r.fleetDBClient.AddServer(c.Request.Context(), serverID, params.Facility, params.IP, params.Username, params.Password); err != nil {
+	if cleanup, err = r.fleetDBClient.AddServer(c.Request.Context(), serverID, params.Facility, params.IP, params.Username, params.Password); err != nil {
+		shouldCleanup = true
```

Comment:
Why not clean up the database record when we discover the error? That is, why push this to the handler instead of doing it at the FleetDB client layer?

Reply:
`FleedDB`client layer doesn't know whether `conditionCreate()` is succeeded or not.
If users failed to create new condition, it looks like users may be able to re-create again? If we don't cleanup, users will get `duplicate` error when enroll server again.

I am not sure if my understanding is correct. I can move the cleanup into the `FleetDB` client layer if that's more appropriate.

Reply:
No, I was slow, but I understand what you're doing here, and you're right: if we fail to create or publish a condition, we need to roll the database change back. Consider a name change of `cleanup` to `rollback` maybe? That would make it clearer to me, but it might not be universal.